src/Runner.Listener/MessageListener.cs

```diff
@@ -170,6 +174,18 @@ public async Task DeleteSessionAsync()
             }
         }
 
+        public void OnJobStarted(object sender, EventArgs e)
+        {
+            runnerStatus = TaskAgentStatus.Busy;
```

Comment:
you might want to add some tracing so we know that the status changed at what time.